ldap_reader/vendor.py

```diff
@@ -100,3 +103,14 @@ def check_enabled(config, attrs_dict):
     '''
     server_type = config["server_type"]
     return _SERVER_ENABLE_ATTR_MAP[server_type][1](attrs_dict)
+
+
+def fix_guid(config, guid):
```

Comment:
I have verified the code against the three servers, and `AD` is the only one that returns the binary blob. 
Regarding RHDS, on my 389-ds setup, `uid` does not work as object UUID field, `nsuniqueid` does.